### PR TITLE
Accumulator Support and Integration Testing

### PR DESCRIPTION
**Pull request by drozdziak1**
This builds on Jayant's WIP PR for accumulator support. Changes:
* Accumulator program binary added to integration tests and passed to agent in test keystore
* log all transaction statuses to DEBUG, log failing ones to WARN

### CODE REVIEW

**Comment by drozdziak1** [comment]
`more updates in tests` as in price updates. The current blocker seems to be tripping the aggregate update, mainly due to the 20-publisher min_pub default. Here's some options:
* mock 20 publishers to meet the default min_pub value for triggering aggregate update and then the CPI
*  hand-roll a `set_min_publishers` transaction to set the default to 1, use a cursed oracle binary that always updates the accumulator just to see how the cross call is doing (might be the easiest)

**Comment by drozdziak1** [comment]
Summary of changes:
* `accumulator_updater.so` - artifact after commenting out whitelisting checks and disabling whitelist account initialization checks
* `oracle.so` - artifact after changing crosscall's `AccountMeta::new_readonly` to `AccountMeta::new` for `accumulator_data` (making it writable for the updater)
*`test_integration.py` - moved the long sleep past assertions, fixed wrongly deploying oracle.so as accumulator

**Comment by drozdziak1** [comment]
Changes:
* updated oracle.so from latest tip of `accumulator-cpi`
* regenerated `message_buffer` with anchorpy 0.14.0 because it contained too new solana/solders usage which clashed with older solana/solders deps in integration test. Added `--pdas` to automate the static whitelist PDA.
* finished calls to initialize() and set_allowed_programs(), logged auth PDA address. Both succeed. **Note: the accumulator takes auth PDAs, not bare program IDs for set_allowed_programs, despite the name** 
* added oracle_auth_pda to exporter.rs oracle call, logged PDA address

Status:
We get a signer escalation error despite passing the correct PDA (as logged in agent, pytest setup and ultimately in `ledger/solana_logs/stdout`).

**Comment by drozdziak1** [comment]
# Finalized summary of changes
This changeset updates Solana program dependencies and implements client code to call into the oracle with accumulator CPI support. The majority of changes accomodates evolving integration testing setup.

## Review highlights
* `integration-tests/message_buffer/` is AnchorPy codegen and can be ignored 

## Agent changes
* Accept a new `<network>.key_store.accumulator_key_path` option, enabling the accumulator CPI in the called oracle
* Exporter: additional create_instruction_with_accumulator implementation containing the correct accounts to perform accumulator CPI, improved failed transaction logging, compute unit limit bumped 20k -> 40k

## Integration tests
* Add message_buffer.so containing the latest accumulator program from pyth-crosschain's `main`
* Add message_buffer.json, the IDL for the *.so
* Add `integration-tests/message_buffer/`, an Anchor client generated with `anchorpy`
* Update oracle.so to the tip of `accumulator-cpi` on `pyth-client`

### `test_integration.py`
* Deploy both oracle and accumulator using `solana-test-validator --bpf-program <addr> <bin.so>` instead of `solana-program-deploy`. This lets us keep the real private key hidden, while keeping the program ID consistent across environments. This also puts Anchor's program ID mismatch errors to rest.
* Add `USE_ACCUMULATOR` env for switching accumulator features on in all parts of the test setup. 
* Add `SOLANA_TEST_VALIDATOR` env for supplying alternative validator binaries - this enables us to use a prebuilt pythnet validator easily and work with its accumulator outputs.
* Update program-admin - this adds the `min_publishers` option on which the test relies for correct aggregate price values
* Fixture a full initialization for the message_buffer program - this does accumulator initialization, complete with permissions and buffer allocation
* Fixture agent config to supply accumulator key when `USE_ACCUMULATOR` is in effect
* Add a disabled test with infinite loop for live experimenting on the running test suite

## Miscellaneous
* Add .dockerignore
* Fail agent startup when config does not exist
* Log working directory and config path in agent

File: src/agent/solana.rs [hunk 1]
```diff
@@ -144,6 +144,8 @@ mod key_store {
         pub program_key_path:     PathBuf,
         /// Path to the public key of the root mapping account, relative to the root
         pub mapping_key_path:     PathBuf,
+        /// Path to the public key of the accumulator program, relative to the root.
+        pub accumulator_key_path: Option<PathBuf>,
```
Comment (jayantk) [inline src/agent/solana.rs:148]: (it's confusing that these pubkeys are in separate files as opposed to simply being strings in the config, but this is consistent with the other keys so it's fine)

**Review by jayantk — APPROVED** [review]
Awesome. I like the toggleable behavior for testing against different environments. My main ask here is for some organization and documentation:

1. write down the env flags in the readme and how/when you should use them.
2. would also be nice to put the binaries into a folder like `program_binaries/`
3. see inline comments about the autogenned python/idl stuff

that's all easy stuff so i'm going to approve this and trust you to fix those issues.

File: integration-tests/tests/test_integration.py [hunk 2]
```diff
@@ -305,12 +297,101 @@ def agent_publish_keypair(self, agent_keystore_path, sync_accounts):
     def agent_keystore(self, agent_keystore_path, agent_publish_keypair):
         self.run(
             f"../scripts/init_key_store.sh localnet {agent_keystore_path}")
+
+        if USE_ACCUMULATOR:
+            path = os.path.join(agent_keystore_path, "accumulator_program_key.json")
+
+            with open(path, 'w') as f:
+                f.write(MESSAGE_BUFFER_PROGRAM)
```
[the remark this answers is not in the document]
Comment (drozdziak1) [reply integration-tests/tests/test_integration.py:305]: Correct, we do this for other addresses too, I think we noticed this on one occasion and decided to remain crappy but consistent

File: src/agent/solana/exporter.rs [hunk 3]
```diff
@@ -128,7 +129,7 @@ impl Default for Config {
             inflight_transactions_channel_capacity:  10000,
             transaction_monitor:                     Default::default(),
             // The largest transactions appear to be about ~12000 CUs. We leave ourselves some breathing room.
-            compute_unit_limit:                      20000,
+            compute_unit_limit:                      40000,
```
[the remark this answers is not in the document]
Comment (drozdziak1) [reply src/agent/solana/exporter.rs:132]: This is configurable, but I agree that we should make sure.

File: integration-tests/message_buffer.json [hunk 4]
```diff
@@ -0,0 +1,573 @@
+{
```
[the remark this answers is not in the document]
Comment (drozdziak1) [reply integration-tests/message_buffer.json:1]: Gathered all artifacts in `program-binaries`, renamed the JSON to `message_buffer_idl.json`

File: src/agent/solana/exporter.rs [hunk 5]
```diff
@@ -390,40 +391,31 @@ impl Exporter {
...
-                        }),
-                    )?,
+            /*
+            let instruction = self.create_instruction_without_accumulator(
+                publish_keypair.pubkey(),
+                Pubkey::new(&identifier.to_bytes()),
+                &price_info,
+                network_state.current_slot,
+            )?;
+             */
```
[the remark this answers is not in the document]
Comment (drozdziak1) [reply src/agent/solana/exporter.rs:401]: Done

File: src/agent/solana/exporter.rs [hunk 6]
```diff
@@ -464,6 +456,138 @@ impl Exporter {
...
+        let (accumulator_data_pubkey, _accumulator_data_pubkey) = Pubkey::find_program_address(
+            &[
+                &oracle_auth_pda.to_bytes(),
+                "message".as_bytes(),
+                &price_id.to_bytes(),
+            ],
+            &accumulator_program_key,
+        );
+
+        debug!(self.logger, "Oracle Auth PDA"; "address" => oracle_auth_pda.to_string());
```
[the remark this answers is not in the document]
Comment (drozdziak1) [reply src/agent/solana/exporter.rs:530]: Removed

**Pull request merged**